### ISSUE TITLE
Subprocess.Popen not working after updating libraries

### ISSUE THREAD

**Issue by akmalhakimi1991** [GoogleCloudPlatform/python-docs-samples#1265]
[arecord.txt](https://github.com/GoogleCloudPlatform/python-docs-samples/files/1547226/arecord.txt)

## In which file did you encounter the issue?

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/speech/cloud-client/transcribe_streaming_mic.py

### Did you change the file? If so, how?

I added few things and change the recording input to arecord (instead of PyAudio). In order to use arecord, I had to use subprocess.Popen to stream the audio and get the input from stdout and send it to Google.

## Describe the issue

This code works fine until a few days ago where I started seeing the "[StatusCode.RESOURCE_EXHAUSTED](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1238)" error message (even though I'm not even closed to exceed my usage quota). I think I was using google-cloud-speech 0.28.0 or 0.29.0. I decided to upgrade this library to the latest version as an attempt to fix the previous issue. However, after updating the libraries, the program would stuck at starting the `subprocess.Popen` call to initiate arecord. It didn't show any error message and it didn't crash the script. I'm not sure why this happened. I tried to uninstall all libraries and reinstall the older version of google-cloud-speech but the same error happened. Is it possible that there's deadlock happening after the update?

Edit: It's not just arecord. I tried with simple command such as `ls` or `pwd` and it didn't work also. However, `os.system` work fine.

**Comment by jonparrott** [comment]
@jerjou @dizcology can on of you take a look?

**Comment by jerjou** [comment]
Hello-

**Re: RESOURCE_EXHAUSTED:**
Have you set your `GOOGLE_APPLICATION_CREDENTIALS` environment variable to the path to your service account (per [this doc](https://cloud.google.com/speech/docs/auth#application_default_credentials))?

**Re: hanging**
Hanging can happen for any number of reasons. Do you have your code anywhere we can take a look at?

**Comment by akmalhakimi1991** [comment]
@jerjou The code is attached. FYI the original sample code (with pyaudio) works fine.

**Comment by akmalhakimi1991** [comment]
To add to this, when I use `p = os.popen("arecord -f S16_LE -r 16000 -t wav")`, I will get "write error" error message

**Comment by akmalhakimi1991** [comment]
To add to this, `os.popen` also works fine:
`p = os.popen("arecord -f S16_LE -r 16000 -t wav")`

**Comment by jerjou** [comment]
Interesting...
okay, so after a couple tests with your code, which sometimes reproduces and sometimes not, i _think_ what might be happening is that the subprocess isn't cleaning itself up cleanly, from previous times I've run the script. I think this because, I was testing this out, and was flummoxed because it seemed like adding `print` statements was making it work. So I switched to doing something else, until my machine started lagging. I ran `top`, and saw that there were still `python arecord.py` processes running, so I killed them. Suspicious, I flipped back to my terminal and ran `python arecord.py` again, and it worked! This is further bolstered because if I run `strace python arecord.py`, the process seems to be blocking on trying to acquire a futuex. So my theory is that there's some shared resource that the previous process is holding on to (because the thread or subprocess didn't terminate cleanly?) that's causing any new process to block, waiting for that resource.

See if that works for you..
